app/user/services.py

```diff
@@ -160,3 +161,78 @@ async def edit_user(user_id: int, data: edit_schemas.UserEdit, db: Session):
     db.commit()
     db.refresh(obj)
     return obj
+
+
+async def create_newsletter_subscriber(email: EmailStr, db: Session):
+    """This function creates a newsletter subscription
+
+    Args:
+        email (str): The subscriber's email
+        db (Session): The database session
+
+    Returns:
+        models.Newsletter: The created newsletter subscription obj
+    """
+    if db.query(models.NewsLetter).filter_by(email=email).first():
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"email {email} already subscribed",
+        )
+    obj = models.NewsLetter(email=email)
+    db.add(obj)
+    db.commit()
+    db.refresh(obj)
+    return obj
+
+
+async def create_company(user_id: int, data: create_schemas.CompanyCreate, db: Session):
+    """This function creates a new company
+
+    Args:
+        data (create_schemas.CompanyCreate): The company's data
+        db (Session): The database session
+
+    Raises:
+        HTTPException[400]: Company with email exists
+
+    Returns:
+        models.Company: The created company obj
+    """
+    # Validate unique email
+    if db.query(models.Company).filter_by(email=data.email).first():
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"company with email {data.email} exists",
+        )
+
+    obj = models.Company(**data.model_dump())
+    obj.user_id = user_id
+    db.add(obj)
+    db.commit()
+    db.refresh(obj)
+    return obj
+
+
+async def edit_company(user_id: int, data: edit_schemas.CompanyEdit, db: Session):
```

Comment:
What if they are changing email or phone or registration to that belonging to another user, it will cause the database to raise an error meaning a 500 will happen